modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/Resolver.scala

```diff
@@ -38,6 +38,13 @@ final case class Resolver[F[_]](repos: List[Registry], cache: Option[ResolverCac
   private[client] val allRepos: NonEmptyList[Registry] =
     NonEmptyList[Registry](Registry.EmbeddedRegistry, repos)
 
+  private val allIgluCentral = repos.collect {
+    case Registry.Http(config, connection)
+        if config.name.toLowerCase.matches(".*iglu[- ]?central.*") && connection.uri.toString
+          .contains("iglucentral") =>
```

Comment:
I think it looks fragile to match on the name.  Because this constrains us to be very careful how we name things in config files.  I think that could very easily break if someone does a well-meaning tidy-up of a config file.

But I do like the match on the uri, because that is a more controlled string.  I guess you could match on `uri.host` instead of `uri.toString`?  And maybe match word boundaries with a regex?

```scala
uri.host.matches(""".*\biglucentral\b.*""")
```

For any other people reading this -- we discussed a lot in a private channel about matching on `iglucentral` in a host name.  We all agree it's not ideal, but we're going to do it anyway because it's the only way to treat the mirrors as a special case, unless we do a bigger refactor.

Reply:
> I think it looks fragile to match on the name. Because this constrains us to be very careful how we name things in config files. I think that could very easily break if someone does a well-meaning tidy-up of a config file.

I wanted to put as much criteria as I could in the filter, but I agree that this is not very future-proof and that the URI is probably enough.

> I guess you could match on uri.host instead of uri.toString

Good idea !

> And maybe match word boundaries with a regex?

It's not as future-proof as without the boundaries, but seems not too dangerous.